File: tipbx/app/src/main/java/com/tipbox/app/fcm/FirebaseInstanceIDService.java
package com.tipbox.app.fcm;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v4.content.LocalBroadcastManager;
import android.util.Log;
import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;
import com.tipbox.app.R;
import com.tipbox.app.util.ServerUrls;


/**
 * Created by vinod on 12/19/2016.
 */
public class FirebaseInstanceIDService extends FirebaseInstanceIdService  {
    private static final String TAG = FirebaseInstanceIDService.class.getSimpleName();

    @Override
    public void onTokenRefresh() {
        super.onTokenRefresh();
        String refreshedToken = FirebaseInstanceId.getInstance().getToken();
        sendRegistrationToServer(refreshedToken);
        Intent registrationComplete = new Intent("registrationComplete");
        registrationComplete.putExtra("token", refreshedToken);
        LocalBroadcastManager.getInstance(this).sendBroadcast(registrationComplete);
    }

    private void sendRegistrationToServer(final String token) {
        Log.e(TAG, "sendRegistrationToServer: " + token);
        storeRegIdInPref(token);
    }

    private void storeRegIdInPref(String token) {
        SharedPreferences pref = getApplicationContext().getSharedPreferences(getString(R.string.app_name), 0);
        SharedPreferences.Editor editor = pref.edit();
        editor.putString(ServerUrls.SET_TOKEN, token);
        editor.commit();
    }
}

File: verify/app/src/main/java/com/wfi/beaconscanner/fcm/FirebaseInstanceIDService.kt
package com.wfi.beaconscanner.fcm

import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService
import com.wfi.beaconscanner.utils.Keys
import com.wfi.beaconscanner.utils.SharedPreferencesManager

/**
 * Created by nisha
 */
class FirebaseInstanceIDService : FirebaseInstanceIdService() {

    override fun onTokenRefresh() {
        super.onTokenRefresh()
        var refreshedToken = FirebaseInstanceId.getInstance().getToken();
        var spm = SharedPreferencesManager(this);
        spm.saveStringValues(Keys.TOKEN,refreshedToken);
    }
}